toonz/sources/toonz/iocommand.cpp

```diff
@@ -2223,6 +2223,163 @@ bool IoCmd::loadScene() {
   }
 }
 //===========================================================================
+// IoCmd::saveSceneVersion()
+//---------------------------------------------------------------------------
+
+
+bool IoCmd::saveSceneAs(const TFilePath &fp) {
+  // copy and replace all levels if project option "Separate assets into scene
+  // sub-folders" is checked
+  TProjectP project = TProjectManager::instance()->getCurrentProject();
+  bool saveTwice    = false;
+  if (project->getUseSubScenePath()) {
+    // save scene a first time before updating levels paths just to make
+    // getDefaultLevelPath method happy (and avoid serach repalace old/new scene
+    // name)...
+    saveTwice   = true;
+    bool result = IoCmd::saveScene(fp, 0);
+
+    ToonzScene *scene   = TApp::instance()->getCurrentScene()->getScene();
+    TLevelSet *levelSet = scene->getLevelSet();
+    for (int i = 0; i < levelSet->getLevelCount(); i++) {
+      TXshLevel *lvl = levelSet->getLevel(i);
+      auto oldPath   = lvl->getPath();
+      auto newPath = scene->getDefaultLevelPath(lvl->getType(), lvl->getName());
+      if (oldPath != newPath) {
+        std::cout << "SaveSceneAsPopup: level oldPath: "
```

Comment:
Please remove these `std::cout` in your code.  These will never be seen unless running command line and doesn't really add any value, except maybe for debugging purposes.

Reply:
hey John,
that's done,
Is message center often used ?
thanks

Reply:
> Is message center often used ?

Not really. I think it might be used more so by the Render Farm. Havent had the need to use Message Center for any particular reason.